internal/gps/result.go

```diff
@@ -61,19 +65,45 @@ func WriteDepTree(basedir string, l Lock, sm SourceManager, sv bool) error {
 		return err
 	}
 
-	// TODO(sdboyer) parallelize
-	for _, p := range l.Projects() {
-		to := filepath.FromSlash(filepath.Join(basedir, string(p.Ident().ProjectRoot)))
+	projects := l.Projects()
+	worksz := len(projects)
+
+	workChan := make(chan LockedProject, worksz)
+	doneChan := make(chan error, worksz)
+
+	for i := 0; i < NumDepTreeWorkers; i++ {
+		go func() {
+			for p := range workChan {
+				to := filepath.FromSlash(filepath.Join(basedir, string(p.Ident().ProjectRoot)))
 
-		err = sm.ExportProject(p.Ident(), p.Version(), to)
+				err := sm.ExportProject(p.Ident(), p.Version(), to)
+				if err != nil {
+					doneChan <- fmt.Errorf("error while exporting %s: %s", p.Ident().ProjectRoot, err)
+					continue
+				}
+
+				if sv {
+					filepath.Walk(to, stripVendor)
+				}
+
+				doneChan <- nil
+			}
+			// TODO(sdboyer) dump version metadata file
+		}()
+	}
+
+	for _, p := range projects {
+		workChan <- p
+	}
+
+	close(workChan)
+
+	for i := 0; i < worksz; i++ {
+		err := <-doneChan
```

Comment:
so, what you are saying is basically this:

we should always wait for all worker routines to finish
if some routine reports an error, note the error and wait for others to finish
return a fmt.Errorf filled with info of all errs (gathered from the tuples received - if any)